pkg/util/ecs/metadata/detection.go

```diff
@@ -137,3 +148,28 @@ func getAgentV4URLFromEnv() (string, error) {
 	}
 	return agentURL, nil
 }
+
+// IsMetadataV4Available checks if the ECS agent version is compatible with the metadata v4 endpoint
+func IsMetadataV4Available(ecsAgentVersion string) (bool, error) {
+	// Agent is deployed directly on EC2 running in ECS
+	currentECSAgentVersion, err := version.NewVersion(ecsAgentVersion)
+	if err != nil {
+		return false, err
+	}
+
+	minimumECSAgentVersion := MinimumECSAgentVersionForMetadataV4OnLinux
+	if runtime.GOOS == "windows" {
+		minimumECSAgentVersion = MinimumECSAgentVersionForMetadataV4OnWindows
+	}
+
+	expectedMinimumECSAgentVersion, err := version.NewVersion(minimumECSAgentVersion)
+	if err != nil {
+		return false, err
+	}
+
+	if currentECSAgentVersion.LessThan(expectedMinimumECSAgentVersion) {
+		return false, fmt.Errorf("ECS agent version %s is less than the minimum required version %s", currentECSAgentVersion, expectedMinimumECSAgentVersion)
```

Comment:
Based on the comment above, where this function is used, I’m not sure this case is an error.
Nothing went wrong. The function successfully parsed the ECS version without any error.

```suggestion
		log.Debugf("ECS agent version %s is less than the minimum required version %s", currentECSAgentVersion, expectedMinimumECSAgentVersion)
		return false, nil
```

Reply:
It makes sense 👍  [c19b427](https://github.com/DataDog/datadog-agent/pull/28007/commits/c19b427667540c571390bc6fd9206da1ccc8e80f)